Tensor construction codemod(ResizeLike) - 1/7

jerryzh168 opened the pull request:
Summary:
Codemod generated with clangr shard mode, 25 files per diff,
motivation: https://github.com/pytorch/pytorch/pull/12407

Differential Revision: D13419563

ezyang commented:
In these diffs can we *please* have a description which explains (1) what the codemod applied is, (2) why we want to apply this codemod and (3) why it is semantics preserving.

ezyang commented on caffe2/core/operator.h:
```diff
@@ -613,6 +616,18 @@ class Operator : public OperatorBase {
     return OperatorBase::template Output<Tensor>(idx, type);
   }
 
+  Tensor* OutputTensorCopyFrom(
```
This looks dead. Did you mean to put this in a different diff?

jerryzh168 replied:
OK, will put in a different diff

ezyang commented:
> In these diffs can we please have a description which explains (1) what the codemod applied is, (2) why we want to apply this codemod and (3) why it is semantics preserving.

Actually, the codemod here was fine; the problem was the changes in caffe2/core/operator.h which were not explained.

jerryzh168 commented:
@ezyang (1) and (2) are in summary((1) is only visible to Facebook)

jerryzh168 commented on caffe2/core/operator.h:
```diff
@@ -208,7 +208,10 @@ class CAFFE2_API OperatorBase : public Observable<OperatorBase> {
+    if (!options.has_dtype()) {
+      options = options.dtype(src.dtype());
+    }
+    Tensor* t = OutputTensor(idx, src.sizes(), options);
```
probably this is for another diff.. I'll split

jerryzh168 committed:
default options for OutputTensorCopyFrom

Summary:
OutputTensorCopyFrom takes four arguments: index, a source Tensor, TensorOptions and whether we want to perform an async call.
We want to provide some default option for TensorOptions, (1). default device to context_.device() (2). default dtype to input.dtype(). User can also explicitly provide these options to override default values.

Differential Revision: D13453824

fbshipit-source-id: a7a5f2b84eb4161125be97842daa2e7f0ecfcc60

jerryzh168 committed:
Tensor construction codemod(ResizeLike) - 1/7 (#15073)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/15073

Codemod generated with clangr shard mode, 25 files per diff,
motivation: https://github.com/pytorch/pytorch/pull/12407

Reviewed By: dzhulgakov

Differential Revision: D13419563

fbshipit-source-id: dfc3624e312abf115c51d49343009e801914e773
